src/cargo/core/resolver/errors.rs

```diff
@@ -327,7 +341,37 @@ pub(super) fn activation_error(
         name_candidates.sort_by_key(|o| o.0);
 
         let mut msg = String::new();
-        if !name_candidates.is_empty() {
+        if !version_candidates.is_empty() {
+            let _ = writeln!(
+                &mut msg,
+                "no matching versions for `{}` found",
+                dep.package_name()
+            );
+            for candidate in version_candidates {
+                match candidate {
+                    IndexSummary::Candidate(summary) => {
+                        // HACK: If this was a real candidate, we wouldn't hit this case.
```

Comment:
Not sure how this happens. It it possible to add a unit test to cover this?

Reply:
I don't know how to hit this. This was originally an assert but I hit it because we switched everything to `Candidate`. That should be fixed but I figured a vague error is better for users than a crash in case I missed something or things change in the future.